app/questionnaire/value_source_resolver.py

```diff
@@ -56,8 +56,14 @@ def _is_block_on_path(self, block_id: str) -> bool:
             # repeating blocks aren't on the path, so check the parent list collector
             list_name = self.schema.list_names_by_list_repeating_block_id[block_id]
             # Type ignore: list collector will exist if the block is repeating
-            list_collector_block: ImmutableDict = self.schema.get_list_collector_for_list(list_name)  # type: ignore
-            return list_collector_block["id"] in self.routing_path_block_ids
+            section: ImmutableDict = self.schema.get_section_for_block_id(block_id)  # type: ignore
+            list_collector_blocks: ImmutableDict = self.schema.get_list_collector_for_list(list_name)  # type: ignore
+
+            for list_collector_block in list_collector_blocks:
+                if section["id"] == self.schema.get_section_id_for_block_id(
+                    list_collector_block["id"]
+                ):
+                    return list_collector_block["id"] in self.routing_path_block_ids
```

Comment:
This doesn't seem quite right. You probably that the method call where you pass in list name and section, instead of fetching all then filtering.